src/Workspaces/Core/Portable/FindSymbols/SymbolTree/SymbolTreeInfo.cs

```diff
@@ -95,6 +90,26 @@ public IEnumerable<ISymbol> Find(IAssemblySymbol assembly, Func<string, bool> pr
             }
         }
 
+        /// <summary>
+        /// Slow, linear scan of all the symbols in this assembly to look for matches.
+        /// </summary>
+        public IEnumerable<ISymbol> Find(IAssemblySymbol assembly, Func<string, bool> predicate, CancellationToken cancellationToken)
+        {
+            for (int i = 0, n = _nodes.Count; i < n; i++)
```

Comment:
It's a habit i have (esp. in JS) for enumerating collections that aren't arrays/strings.  We don't get any jit optimizations here, so i just do the manual optimization.  Each .Count has to go through an interface dispatch as well.  So, versus just getting the value and comparing it is expensive (relatively).